cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs

```diff
@@ -1735,10 +1736,35 @@ instance Show vNumber => ToJSON (HandshakeException vNumber) where
                  ]
 
 instance ToJSON NodeToNodeVersion where
-  toJSON x = String (pack $ show x)
+  toJSON NodeToNodeV_7  = Number 7
+  toJSON NodeToNodeV_8  = Number 8
+  toJSON NodeToNodeV_9  = Number 9
+  toJSON NodeToNodeV_10 = Number 10
+
+instance FromJSON NodeToNodeVersion where
+  parseJSON (Number 7) = return NodeToNodeV_7
+  parseJSON (Number 8) = return NodeToNodeV_8
+  parseJSON (Number 9) = return NodeToNodeV_9
+  parseJSON (Number 10) = return NodeToNodeV_10
+  parseJSON (Number x) = fail ("unsupported version " ++ show x)
```

Comment:
I think `"Unsupported node version " ++ ..` would be better

Reply:
or even `node-to-node protocol version` :)

Reply:
Fixed.